app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php

```diff
@@ -379,13 +380,13 @@ private function addFilteringByChildProductsToSelect(Select $select, Store $stor
             []
         )->joinLeft(
             ['child_cpsd' => $this->getTable('catalog_product_entity_int')],
-            'child_cpsd.' . $linkField . ' = '. 'relation_product_entity.' . $linkField
+            'child_cpsd.' . $linkField . ' = ' . 'relation_product_entity.' . $linkField
```

Comment:
is there a safer alternative to this?
just concatenating strings as parameters to a mysql query could lead to a sql injection